src/components/GroupSystems/GroupImmutableSystems.js

```diff
@@ -187,7 +187,7 @@ const GroupImmutableSystems = ({ groupName, groupId, ...props }) => {
 
   //enable disable bulk update based on selection, must refactor
   useEffect(() => {
-    if (selected.size > 0) {
+    if (selected.size > 0 && deviceImageSet?.size > 0) {
```

Comment:
Shouldn't `deviceImageSet` be added as a dependency to the useEffect's dependency array ? 🤔 
The condition could change when it is changed, which would effect the Effect

Reply:
@LiorKGOW included the validation on useEffect but I kept the condition cause we should not get in this validation if there is no imageSet